server-config.go

```diff
@@ -118,19 +119,32 @@ func (c *configV2) JSON() string {
 	return string(loggerBytes)
 }
 
+// configPath for custom config path only for testing purposes
+var customConfigPath string
+
+func setConfigPath(configPath string) {
+	customConfigPath = configPath
+}
+
 // getConfigPath get users config path
 func getConfigPath() (string, *probe.Error) {
 	if customConfigPath != "" {
 		return customConfigPath, nil
 	}
-	u, err := userCurrent()
-	if err != nil {
-		return "", err.Trace()
+	homeDir, e := user.HomeDir()
+	if e != nil {
+		return "", probe.NewError(e)
 	}
-	configPath := filepath.Join(u.HomeDir, ".minio")
+	configPath := filepath.Join(homeDir, ".minio")
 	return configPath, nil
```

Comment:
If user package does best attempt to find homedir and user, this function can remain as is, I mean simpler like this. We can invoke this function in syntax-check for main and verify if we are OK to proceed.  If user has not set config folder and if we are not able to auto-determine, then we fail.